src/dmd/cparse.d

```diff
@@ -2318,25 +2327,35 @@ final class CParser(AST) : Parser!AST
                         {
                             error("static array parameters are not supported");
                         }
-
-                        if (token.value == TOK.rightBracket)
+                        if (declarator != DTR.xparameter)
                         {
-                            // An array of unknown size, fake it with a DArray
-                            ta = new AST.TypeDArray(t); // []
-                            nextToken();
+                            /* C11 6.7.6.2-4: '*' can only be used with function prototype scope.
+                             */
+                            if (isVLA)
+                                error("variable length array used outside of function prototype");
```

Comment:
This condition isn't quite right, need to also check `level`, but that isn't available to lookup here.  However, VLA's aren't supported, so it's not really a worry.